### PR TITLE
refactored the code space, abstracted one more layer for all UI layer to use

### PR DESCRIPTION
**Pull request by JunqiZhang0**
1. Refactored the code space, abstracted one more layer for all UI layer to use
    2. Changed the golangci-lint github action to @v2 since @v1 has bug
       sometimes
    3. change all imports for local import to direct imports(easy to migrate the project later)
    4. Added tests

### CODE REVIEW

**Comment by JunqiZhang0** [comment]
one of the golang naming convention, https://medium.com/better-programming/naming-conventions-in-go-short-but-descriptive-1fa7c6d2f32a

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,9 @@
-module github.com/abraverm/openengine
+module openengine
 
 go 1.14
 
 require (
+	github.com/abraverm/openengine v0.0.0-20201027075718-848cd7d650eb
```
Comment (abraverm) [inline go.mod]: what is the point of changing "github.com/abraverm/openengine" to "openengine" when you need to add this line with a specific commit as a requirement?

File: restapi/restapiserver.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Package restapi is where all restapi go
+package restapi
```
Comment (abraverm) [inline restapi/restapiserver.go]: I think this file should be part of other PR.

**Review by abraverm — COMMENTED** [review]
Removing all the tests lowered the coverage ratio by 15%

File: go.mod (hunk 1 above)
Comment (JunqiZhang0) [reply go.mod]: This was automatically added by the IDE, I'll see if I can remove this later when this pr is done

File: restapi/restapiserver.go (hunk 2 above)
Comment (JunqiZhang0) [reply restapi/restapiserver.go]: OK, will remove it later

**Comment by JunqiZhang0** [comment]
> Removing all the tests lowered the coverage ratio by 15%

 This PR is not done yet, I 'm still working on improve the code coverage

File: .golangci.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ run:
   tests: false
 linters:
   enable-all: true
-  disable:
+  disable:
```
Comment (abraverm) [inline .golangci.yml]: typo

File: coverage.txt [hunk 4]
```diff
@@ -0,0 +1,334 @@
+mode: atomic
```
Comment (abraverm) [inline coverage.txt]: please add `**/coverage.txt` to gitignore

**Review by abraverm — COMMENTED** [review]
Looks good overall

**Review by abraverm — APPROVED** [review]

**Pull request closed without merging**